core/arch/arm/mm/tee_pager.c

```diff
@@ -181,7 +182,10 @@ static uint32_t get_area_mattr(struct tee_pager_area *area __unused)
 	uint32_t attr = TEE_MATTR_VALID_BLOCK | TEE_MATTR_GLOBAL |
 			TEE_MATTR_CACHE_DEFAULT | TEE_MATTR_SECURE;
 
-	attr |= TEE_MATTR_PRWX;
+	if (area->store)
```

Comment:
As `area` is used now, please remove the `__unused` tag.

Base the attributes on `area->flags` instead, something like:
```
	attr |= TEE_MATTR_PR;
	if (!(area->flags & TEE_PAGER_AREA_RO))
		attr |= TEE_MATTR_PW;
	if (area->flags & TEE_PAGER_AREA_X)
		attr |= TEE_MATTR_PX;
```